src/main/java/de/ustutt/iaas/cc/core/RemoteTextProcessorMulti.java

```diff
@@ -13,18 +21,30 @@
  */
 public class RemoteTextProcessorMulti implements ITextProcessor {
 
+	
+	// TODO Insert ConcurrentLinkedQueue for concurrency optimization
+	private ConcurrentLinkedQueue<String> textProcessorRessources;
+	private Client client;
+	private final static Logger logger = LoggerFactory.getLogger(RemoteTextProcessorMulti.class);
+	
 	public RemoteTextProcessorMulti(List<String> textProcessorResources, Client client) {
 		super();
-		// TODO ...
+		this.textProcessorRessources = new ConcurrentLinkedQueue<String>(textProcessorResources);
+		this.client = client;
 	}
 
 	@Override
 	public String process(String text) {
-		// text processing placeholder, to be replaced by your solution :-)
-		String processedText = "[processed by incomplete RemoteTextProcessorMulti] - " + text;
-
-		// TODO send request to "next" text processor endpoint (following some load balancing strategy)
-
+				
+		String ressource = textProcessorRessources.poll();		

```

Comment:
Removing items from the list will, when running this in a scenario with massive parallel requests, very fast result in an empty list.

If you want to use a list for this, you don't need to synchronize each single access to the list but the combination of poll() and add(), i.e. you would need to do something like this:

```
synchronized(this){  
  resource = resources.poll();  
  // call the service
  resources.add(resource);
}
```

But, this will then again be a bottleneck, as the service call is inside the synchronized block, i.e. you can only execute one service call at a time. I think that the approach of removing and then adding-back endpoints from and to a list will not work here.